src/pages/SignUp.jsx

```diff
@@ -8,59 +8,66 @@ import { Formik, Field, Form, ErrorMessage } from 'formik';
 import { Link, useNavigate } from "react-router-dom";
 import {signUpInputNames, signUpInputsInitialValues } from "../utils/constants.js";
 import { signUpSchema } from "../utils/formSchema.js";
-import { useAppProvider } from "../context/AppContext";
+import { useSelector, useDispatch } from "react-redux";
+import { sessionsActions } from "../store/sessions-slice";
+import { loginSignUpActions as signUpActions } from "../store/login-signUp-slice";
 
 const SignUp = () => {
-    const {
-        updateSessions,
-        showSignUpErrorMsg,
-        setShowSignUpErrorMsg,
-        passwordNotRepeatedError,
-        setPasswordNotRepeatedError,
-        saveLogged,
-    }  = useAppProvider();
+    const dispatch = useDispatch();
+    const signUpErrors = useSelector(state => state.loginSignUp.signUpErrors);
     const navigateTo = useNavigate();
     useEffect(() => {
-        setShowSignUpErrorMsg({ show: false, existingPart: ""});
-    }, [setShowSignUpErrorMsg]);
+        dispatch(signUpActions.setSignUpErrors({
+            passwordNotRepeated: false,
+            accountAlreadyExist: { show: false, existingPart: ""}
+        }));
+    }, [dispatch]);
 
     return (
         <div className="sign-up">
             <Icon name="logo" iconClassName="sign-up__logo" />
             <Formik
                 initialValues={signUpInputsInitialValues}
                 validationSchema={signUpSchema}
-                onSubmit={(signInInfo) => {
+                onSubmit={(signInInfo, actions) => {
                     let passwordRepeatedCorrectly = signInInfo.password === signInInfo.repeatPassword;
                     if(passwordRepeatedCorrectly) {
                         let isANewUser = isANewAccount(signInInfo);
                         if(isANewUser.error) {
-                            console.log("failure");
-                            setShowSignUpErrorMsg({
-                                show: true,
-                                invalidParameter: `${isANewUser.invalidParameter}`,
-                            })
+                            dispatch(signUpActions.setSignUpErrors({
+                                ...signUpErrors,
+                                accountAlreadyExist: {
+                                    show: true,
+                                    invalidParameter: `${isANewUser.invalidParameter}`,
```

Comment:
if the whole string intepolation is just a variable, there's no need to use string interpolation